# Gene-level-frameshifting
**Scripts to analyze gene level frameshifting from Ribo-seq dataset**
****
**The first script: frameshift.py calculates the p-site offset for each aligned bam file (genomic coordinates alignment) and uses the p-site offsets to calculate the frame of each read and outputs a csv file. The script uses a bed annotation file for the CDS to limit the analysis to the coding sequence. This bed annotation file can be generated by the script create_annotation.py**
****
**The next set of scripts are focused on identifying frameshift location. The order of running is:
1- frameshift_loci.py, 2- post_fs_contex.py, 3- post_fs_summary.py, 4- fs_feature_deepdive.py, 5- fs_impact_analyzer.py, The output files were used to generate the data and figures in the paper and train the machine learning algorithm which was performed in R language **
